airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -225,7 +225,7 @@ def get_connection(self, conn_id: str) -> Connection | None:
 
         uri = response.get("conn_uri")
         if uri:
-            return Connection(conn_id, uri=uri)
+            return Connection.from_uri(conn_id=conn_id, uri=uri)
```

Comment:
Oh.. I think we need to solve it in backcompat way 😞

Reply:
Oh yeah, provider code, forgot